manager/controlapi/ca_rotation.go

```diff
@@ -0,0 +1,216 @@
+package controlapi
+
+import (
+	"bytes"
+	"context"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+
+	"github.com/cloudflare/cfssl/helpers"
+	"github.com/docker/swarmkit/api"
+	"github.com/docker/swarmkit/ca"
+	"github.com/docker/swarmkit/log"
+)
+
+var minRootExpiration = helpers.OneYear
+
+// Creates and updates the api.RootRotation object and the cross-signed intermediate.
+// This function assumes that a root rotation is not already in progress, and that the root
+// cert and key to be rotated in or the external CAs in the cluster have already been validated
+func updateRootRotationObject(ctx context.Context, securityConfig *ca.SecurityConfig, cluster *api.Cluster, newRootCA ca.RootCA, version uint64) error {
```

Comment:
The `api` module is just data, and I don't know if we should be adding this to that module, particular could create import issues since then nothing from `ca` would be able to import anything from `api`, since `api` would then depend in `ca`.